### PR TITLE
Feature: configuration endpoints

### PR DESCRIPTION
**Pull request by aboubakar14**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -129,46 +129,46 @@ class Factory:
 
 ### Example of configuration
 
-Finally, you need to add the new AI in the `configuration.json` file.
+Finally, you will have a `configuration` similar to:
```
Comment (Zorin95670) [inline CONTRIBUTING.md:132]: you have to explain how to create the new configuration file...

Comment (aboubakar14) [reply CONTRIBUTING.md:132]: done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -129,46 +129,97 @@ class Factory:
...
+
+- **`key`**  
+  - **Purpose**: A unique identifier for the field used internally by the system.  
+  - **Use**: Represents the configuration parameter (e.g., `base_url`, `defaultModel`).
+
+- **`type`**  
+  - **Purpose**: Specifies the type of input expected in the UI.  
+  - **Values**:  
+    - `"text"`: Single-line text input.  
+    - `"checkbox"`: Toggleable option (true/false).
```
Comment (Zorin95670) [inline CONTRIBUTING.md]: there is no "checkbox" replace it by select

Comment (aboubakar14) [reply CONTRIBUTING.md]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**